### PR TITLE
feat: add actions slot on image block and avatar components

### PR DESCRIPTION
**Pull request by matmkian**
BREAKING CHANGE:
No more tags prop on image block -> now a transclude slot

Need React part

### CODE REVIEW

File: src/components/avatar/react/Avatar.tsx [hunk 1]
```diff
@@ -67,7 +68,9 @@ const DEFAULT_PROPS: IDefaultPropsType = {
  * @return The component.
  */
 const Avatar: React.FC<AvatarProps> = ({
+    actions,
     className = '',
+    hasActions = DEFAULT_PROPS.hasActions,
```
Comment (gcornut) [inline src/components/avatar/react/Avatar.tsx]: Does `hasActions` has to be a props? If you check that there is actions in the `actions` props you would know if there is actions, right?
Also make sure to document the props in the `IAvatarProps` interface in this file

File: src/components/image-block/react/ImageBlock.tsx [hunk 2]
```diff
@@ -97,13 +99,15 @@ const DEFAULT_PROPS: IDefaultPropsType = {
...
 const ImageBlock: React.FC<ImageBlockProps> = ({
+    actions,
     align = DEFAULT_PROPS.align,
     aspectRatio = DEFAULT_PROPS.aspectRatio,
     className = '',
     captionPosition = DEFAULT_PROPS.captionPosition,
     captionStyle = DEFAULT_PROPS.captionStyle,
     description = DEFAULT_PROPS.description,
     fillHeight = DEFAULT_PROPS.fillHeight,
+    hasActions = DEFAULT_PROPS.hasActions,
```
Comment (gcornut) [inline src/components/image-block/react/ImageBlock.tsx]: Same as for the avatar

File: demo/react/components/avatar/default.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React, { CSSProperties, ReactElement } from 'react';
 
-import { Avatar, Size, Theme } from 'LumX';
+import { Avatar, ButtonEmphasis, IconButton, Size, Theme } from 'LumX';
+import { mdiDelete, mdiEye, mdiPencil } from 'LumX/icons';
 
 const demoContainerStyle: CSSProperties = {
```
Comment (gcornut) [inline demo/react/components/avatar/default.tsx]: If the MDX demo is up to date you can remove the TSX demo.
It's an old format of demo that we want to migrate to MDX

File: src/components/avatar/react/Avatar.tsx (hunk 1 above)
Comment (Mcourt) [reply src/components/avatar/react/Avatar.tsx]: I was wondering about that. I looked around the design system and saw that the CommentBlock component has this behavior. I thought maybe it was used to show / hide the actions without changing the "action" props.

Will remove :)

File: src/components/image-block/react/ImageBlock.tsx (hunk 2 above)
Comment (Mcourt) [reply src/components/image-block/react/ImageBlock.tsx]: Yeah I agree it doesn't have sense here. Was about the commit a change

File: demo/react/components/avatar/default.tsx (hunk 3 above)
Comment (Mcourt) [reply demo/react/components/avatar/default.tsx]: Ah ! Was wondering about that. Thank, will remove :)

File: src/components/image-block/react/ImageBlock.tsx [hunk 4]
```diff
@@ -42,8 +42,8 @@ interface IImageBlockProps extends IGenericProps {
     fillHeight?: boolean;
     /* The url of the image we want to display in the image-block. */
     image: string;
-    /** A list of tags, those tags will be displayed in a chip component. */
-    tags?: string[];
+    /* Tags elements to be transcluded into the component */
```
Comment (gcornut) [inline src/components/image-block/react/ImageBlock.tsx]: Could you replace the `/*` comments with `/**` jsdoc comments in this interface and the avatar props interface?
Normal comments are not detected by the program generating the properties table on the demo site.
You can make sur the property tables appear by running with `yarn start:react` and naviguating to the avatar and imageblock demo page (in product > components)

File: demo/react/doc/product/components/avatar.mdx [hunk 5]
```diff
@@ -16,7 +17,44 @@ Comes in 5 sizes: xs, s, m, l, xl.
...
+                            />
+                        </div>
+                    </div>
+                }
+            />
     </>
 );
-```
\ No newline at end of file
+```
```
Comment (gcornut) [inline demo/react/doc/product/components/avatar.mdx]: Could you add the following at the end to show the component properties table?
```
### Properties

<PropTable component="Avatar" />
```

File: CHANGELOG.react.md [hunk 6]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 -   TypeScript and JSDoc cleanup for `SideNavigation` component.
 -   _[BREAKING]_ The `label` and `helper` of the `RadioButton` component are now props and not children.
 -   _[BREAKING]_ The `onChange` callback of the `RadioButton` component now provides the value directly without wrapping object.
+-   _[BREAKING]_ The `tags` prop of the `ImageBlock` must now be JSX and not an array.
```
Comment (gcornut) [inline CHANGELOG.react.md]: :+1:

**Pull request merged**